tar/storage/storage_read.c

```diff
@@ -653,6 +686,27 @@ callback_cache_free(void * record, void * cookie)
 	return (0);
 }
 
+/**
+ * storage_read_cache_free(cache):
+ * Free the cache ${cache}.
+ */
+static void
+storage_read_cache_free(struct storage_read_cache * cache)
+{
+
+	/* Behave consistently with free(NULL). */
+	if (cache == NULL)
+		return;
+
+	/* Free contents of cache. */
+	if (cache->cache_ht != NULL)
```

Comment:
Is it possible for `cache_ht` to be NULL here?  I prefer to avoid NULL checks if possible since they can mask bugs.

Reply:
> Is it possible for cache_ht to be NULL here?

Yes-ish.  I mean, with the current setup of `storage_read_cache_init()`, it's possible.

* `cache` is allocated
* `rwhashtab_init()` fails for some reason.  (`malloc` failing, or `crypto_entropy_read()`, or `keylength == 0`)
* we jump to `err1` in `storage_read_cache_init()`, which currently calls `storage_read_cache_free()`.

It should be safe to replace the _init function's `err1` with a `free()` instead; I've pushed a new commit to do that.